modules/@apostrophecms/admin-bar/ui/apos/components/TheAposContextBar.vue

```diff
@@ -483,7 +485,7 @@ export default {
       apos.bus.$emit('refreshed');
     },
     async onDiscardDraft(e) {
-      const result = await this.discardDraft(this.action, this.context._id, !!this.context.lastPublishedAt);
+      const result = await this.discardDraft(this.context);
```

Comment:
This pattern covers more use cases with less code, I picked up the idea from Stu and ran with it.